### PR TITLE
Add an article project describing how to use Application Insights from Batch Nodes and Tasks

### PR DESCRIPTION
**Pull request by paselem**

### CODE REVIEW

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/App.config [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/App.config]: The folder `Microsfot` is misspelled.

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/ApplicationInsights.config [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      Use the following syntax here to collect additional performance counters:
+      
+      <Counters>
+        <Add PerformanceCounter="\Process(??APP_WIN32_PROC??)\Handle Count" ReportAs="Process handle count" />
+        ...
+      </Counters>
+      
+      PerformanceCounter must be either \CategoryName(InstanceName)\CounterName or \CategoryName\CounterName
+      
+      NOTE: performance counters configuration will be lost upon NuGet upgrade.
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/ApplicationInsights.config]: Does this include me when I upgrade packages dependencies in the samples repo?

Is there a way to avoid this?

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.Batch.Samples.TelemetryInitializer;
+using System;
+
+namespace Microsfot.Azure.Batch.Samples.TelemetryStartTask
+{
+    class Program
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: public class

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        static void Main(string[] args)
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: public static...

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.ApplicationInsights.Extensibility;
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: Do the other samples/article projects have usings inside or outside the namespace? From what I can see it looks like we've got it inside for most samples? Update yours to be inside as well?

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Properties/AssemblyInfo.cs]: This whole file is unnecessary, you can just delete it.

File: CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/ApplicationInsights.config [hunk 7]
```diff
@@ -0,0 +1,49 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/ApplicationInsights.config]: Duplicate file? I guess maybe the one with the spelling mistake can just be deleted?

File: CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/Properties/AssemblyInfo.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/Properties/AssemblyInfo.cs]: Delete this file

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config [hunk 9]
```diff
@@ -0,0 +1,61 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+	<configSections>
+  <sectionGroup name="userSettings" type="System.Configuration.UserSettingsGroup, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089">
+   <section name="Microsoft.Azure.Batch.Samples.TopNWordsSample.Settings" type="System.Configuration.ClientSettingsSection, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" allowExeDefinition="MachineToLocalUser" requirePermission="false" />
+  </sectionGroup>
+ </configSections>
+ <startup>
+		<supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config]: wonky tabs

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+    /// each input blob.
+    /// </summary>
+    public static class Job
+    {
+        // files that are required on the compute nodes that run the tasks
+        private const string TopNWordsExeName = "TopNWordsSample.exe";
+        private const string StorageClientDllName = "Microsoft.WindowsAzure.Storage.dll";
+
+        // Application Insights assemblies
+        private const string AIConfig = "ApplicationInsights.config";
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: You could probably accomplish this more cleanly with 
```
private static readonly List<string> AssembliesToUpload = new List<string> 
{
    "Microsoft.ApplicationInsights.dll",
    ...
}
```

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+                var resourceHelperTask = SampleHelpers.UploadResourcesAndCreateResourceFileReferencesAsync(
+                    cloudStorageAccount,
+                    AIBlobConatinerName,
+                    files);
+
+                List<ResourceFile> resourceFiles = resourceHelperTask.Result;
+
+                pool.StartTask = new StartTask()
+                {
+                    CommandLine = string.Format("cmd /c {0}", BatchStartTaskTelemetryRunnerName),
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: minor: Could use string interpolation throughout this file I think

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+
+        // Batch start task telemetry runner
+        private const string BatchStartTaskFolderName = "StartTask";
+        private const string BatchStartTaskTelemetryRunnerName = "Microsfot.Azure.Batch.Samples.TelemetryStartTask.exe";
+        private const string BatchStartTaskTelemetryRunnerAIConfig = "ApplicationInsights.config";
+
+        // Container where monitoring assemblies are placed
+        private const string AIBlobConatinerName = "batchmonitoringassemblies";
+
+        public static void JobMain(string[] args)
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: Why not make this async?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+                    CommandLine = string.Format("cmd /c {0}", BatchStartTaskTelemetryRunnerName),
+                    ResourceFiles = resourceFiles
+                };
+
+                Console.WriteLine("Adding pool {0}", topNWordsConfiguration.PoolId);
+                try
+                {
+                    GettingStartedCommon.CreatePoolIfNotExistAsync(client, pool).Wait();
+                }
+                catch (AggregateException ae)
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: Yeah definitely make this async, you dodge so much pain

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 14]
```diff
@@ -0,0 +1,337 @@
...
+                try
+                {
+                    Console.WriteLine("Creating job: " + topNWordsConfiguration.JobId);
+                    // get an empty unbound Job
+                    CloudJob unboundJob = client.JobOperations.CreateJob();
+                    unboundJob.Id = topNWordsConfiguration.JobId;
+                    unboundJob.PoolInformation = new PoolInformation() { PoolId = topNWordsConfiguration.PoolId };
+
+                    // Commit Job to create it in the service
+                    unboundJob.Commit();
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: commitasync

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 15]
```diff
@@ -0,0 +1,337 @@
...
+                    // by Storage which will lengthen the overall processing time.
+                    //
+                    // You'll need to observe the behavior and use published techniques for finding the right balance of performance versus
+                    // complexity.
+
+                    string[] documents = Directory.GetFiles(topNWordsConfiguration.DocumentsRootPath);
+                    List<string> documentUris = new List<string>();
+                    foreach (string document in documents)
+                    {
+                        documentUris.Add(UploadFileToCloudBlob(accountSettings, document));
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: Weird that you are using `FileToStage` in some cases but not others. What's the reasoning?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 16]
```diff
@@ -0,0 +1,337 @@
...
+        private static CloudBlobClient GetCloudBlobClient(string accountName, string accountKey, string accountUrl)
+        {
+            StorageCredentials cred = new StorageCredentials(accountName, accountKey);
+            CloudStorageAccount storageAccount = new CloudStorageAccount(cred, accountUrl, useHttps: true);
+            CloudBlobClient client = storageAccount.CreateCloudBlobClient();
+
+            return client;
+        }
+
+        private static string booksContainerName = "books";
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: Move this to the top? And make it a const?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs [hunk 17]
```diff
@@ -0,0 +1,337 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Upload a text file to a cloud blob.
+        /// </summary>
+        /// <param name="accountSettings">The account settings.</param>
+        /// <param name="fileName">The name of the file to upload</param>
+        /// <returns>The URI of the blob.</returns>
+        private static string UploadFileToCloudBlob(AccountSettings accountSettings, string fileName)
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: does a helper for this not already exist?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWords.csproj [hunk 18]
```diff
@@ -0,0 +1,201 @@
...
+    <Compile Include="Program.cs" />
+    <Compile Include="Settings.Designer.cs">
+      <AutoGen>True</AutoGen>
+      <DesignTimeSharedInput>True</DesignTimeSharedInput>
+      <DependentUpon>Settings.settings</DependentUpon>
+    </Compile>
+    <Compile Include="TopNWordsTask.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="documents\180555main_ETM.Teachers.Guide.txt">
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWords.csproj]: Have you confirmed that it's safe to use these files?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+            }
+
+            string blobName = args[1];
+            int numTopN = int.Parse(args[2]);
+            string storageAccountName = args[3];
+            string storageAccountKey = args[4];
+
+            WordCount wordCounter = new WordCount(insightsClient);
+            wordCounter.CountWords(blobName, numTopN, storageAccountName, storageAccountKey);
+            wordCounter.Dispose();
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: use `using` -- don't call `dispose` directly

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+        }
+    }
+
+    public class WordCount
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: Shouldn't this implement `IDisposable` if you're implementing `Dispose`?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+                    this.commonProperties.Add("TaskId", Environment.GetEnvironmentVariable("AZ_BATCH_TASK_ID"));
+                    this.commonProperties.Add("PoolId", Environment.GetEnvironmentVariable("AZ_BATCH_POOL_ID"));
+                    this.commonProperties.Add("NodeId", Environment.GetEnvironmentVariable("AZ_BATCH_NODE_ID"));
+                }
+
+                return this.commonProperties;
+            }
+        }
+
+        private TelemetryClient insightsClient;
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: minor: Move these to the top of the class?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+
+        public WordCount(TelemetryClient telemetryClient)
+        {
+            this.insightsClient = telemetryClient;
+            this.taskId = Environment.GetEnvironmentVariable("AZ_BATCH_TASK_ID");
+        }
+
+        public void CountWords(string blobName, int numTopN, string storageAccountName, string storageAccountKey)
+        {
+            // simulate exception
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: Clarify this comment some? Why are you doing this?

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+            using (Stream memoryStream = new MemoryStream())
+            {
+                // Find blob download time
+                DateTime start = DateTime.Now;
+                blob.DownloadToStream(memoryStream);
+                TimeSpan downloadTime = DateTime.Now.Subtract(start);
+                insightsClient.TrackMetric("Blob download in seconds", downloadTime.TotalSeconds, this.CommonProperties);
+
+                memoryStream.Position = 0; //Reset the stream
+                var sr = new StreamReader(memoryStream);
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: Missing a using here

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+            }
+
+            // open the cloud blob that contains the book
+            var storageCred = new StorageCredentials(storageAccountName, storageAccountKey);
+            CloudBlockBlob blob = new CloudBlockBlob(new Uri(blobName), storageCred);
+            using (Stream memoryStream = new MemoryStream())
+            {
+                // Find blob download time
+                DateTime start = DateTime.Now;
+                blob.DownloadToStream(memoryStream);
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: Why download it to memory rather than a file?

File: CSharp/Common/AccountSettings.settings [hunk 25]
```diff
@@ -1,9 +1,9 @@
 ﻿<?xml version='1.0' encoding='utf-8'?>
-<SettingsFile xmlns="http://schemas.microsoft.com/VisualStudio/2004/01/settings" CurrentProfile="(Default)" GeneratedClassNamespace="Microsoft.Azure.Batch.Samples.Common" GeneratedClassName="Settings">
+<SettingsFile xmlns="http://schemas.microsoft.com/VisualStudio/2004/01/settings" CurrentProfile="(Default)" GeneratedClassNamespace="Microsoft.Azure.Batch.Samples.Common" GeneratedClassName="AccountSettings">
```
Comment (matthchr) [inline CSharp/Common/AccountSettings.settings]: No changes so revert this file?

File: CSharp/Common/AccountSettings.Designer.cs [hunk 26]
```diff
@@ -1,7 +1,7 @@
 ﻿//------------------------------------------------------------------------------
 // <auto-generated>
 //     This code was generated by a tool.
-//     Runtime Version:4.0.30319.18444
+//     Runtime Version:4.0.30319.42000
 //
 //     Changes to this file may cause incorrect behavior and will be lost if
```
Comment (matthchr) [inline CSharp/Common/AccountSettings.Designer.cs]: No changes so revert this file?

File: CSharp/BatchExplorer/Models/TaskModel.cs [hunk 27]
```diff
@@ -103,7 +103,7 @@ public bool HasOutputFiles
                 if (this.attemptToLoadOutputs && (this.OutputFiles == null || !this.OutputFiles.Any()))
                 {
                     AsyncOperationTracker.Instance.AddTrackedInternalOperation(
```
Comment (matthchr) [inline CSharp/BatchExplorer/Models/TaskModel.cs]: Why did you change this?

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/App.config (hunk 1 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/App.config]: done

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/ApplicationInsights.config (hunk 2 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/ApplicationInsights.config]: No, there is unfortunately no way to avoid this. The only thing is to not update the application insights packages. Other package changes should not affect this.

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs (hunk 5 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: It's actually mixed... I updated to put them all inside the namespace for this project though.

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs (hunk 3 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs (hunk 4 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Program.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsfot.Azure.Batch.Samples.TelemetryStartTask/Properties/AssemblyInfo.cs]: done.

File: CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/ApplicationInsights.config (hunk 7 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/ApplicationInsights.config]: Not a dupe. It is there intentionally.

File: CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/Properties/AssemblyInfo.cs (hunk 8 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/Microsoft.Azure.Batch.Samples.TelemetryInitializer/Properties/AssemblyInfo.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config (hunk 9 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config]: fixed

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 10 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 16 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 14 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 12 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 13 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs (hunk 17 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs]: done - updated to use sample helpers

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 20 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 19 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 21 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 22 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 23 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: done

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs (hunk 24 above)
Comment (paselem) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/TopNWordsTask.cs]: No reason. Since the files are small, there really isn't any use to download it to disk just to load it up into memory again.

File: CSharp/Common/AccountSettings.Designer.cs (hunk 26 above)
Comment (paselem) [inline CSharp/Common/AccountSettings.Designer.cs]: done

**Comment by matthchr** [comment]
I have approved, but do not merge until Pablo gives the ok that the files have been approved for use.

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config [hunk 28]
```diff
@@ -45,13 +45,13 @@
         <value>True</value>
       </setting>
       <setting name="JobId" serializeAs="String">
-        <value>TopNWordsJob</value>
+        <value>TopNWordsJob2</value>
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/App.config]: This seems wrong

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Program.cs [hunk 29]
```diff
@@ -14,7 +16,7 @@ public static void Main(string[] args)
             }
             else
             {
-                Job.JobMain(args);
+                Task.Run(() => Job.JobMain(args)).GetAwaiter().GetResult();
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Program.cs]: Don't think you need `Task.Run` here.

File: CSharp/ArticleProjects/ApplicationInsights/TopNWords/Settings.settings [hunk 30]
```diff
@@ -21,13 +21,13 @@
       <Value Profile="(Default)">True</Value>
     </Setting>
     <Setting Name="JobId" Type="System.String" Scope="User">
-      <Value Profile="(Default)">TopNWordsJob</Value>
+      <Value Profile="(Default)">TopNWordsJob2</Value>
```
Comment (matthchr) [inline CSharp/ArticleProjects/ApplicationInsights/TopNWords/Settings.settings]: Also seems wrong?

**Comment by matthchr** [comment]
@paselem Pinging on this -- there are a few open items still
